mantidproject/mantid
# Muon peak finding algorithm

**Description of work.**

Takes a Workspace2D as input.
Fits a background of type specified by the user (available: flat, linear, exponential, gaussian and stretched exponential), if no parameters are provided it will attempt to guess them.
Remove the background and apply wiener filter and averaging over neighboring values. The parameters for both can be set by the user.
Find all the peaks in the data and rank them by prominence (see scipy.signal.find_peaks).
Add a number of gaussian peaks in a range specified by the user and return the number of peaks that minimizes the chi squared of the overall fitting function. If Chi2Tolerance is non-zero it will return the highest number of peaks whose chi square value is less that Chi2Tolerance% higher than the lowest value.
The centre, height, width of the peaks are returned as a table.
The raw data and the peaks are plotted to allow for easier tweaking of the parameters to obtain better results (this can be disabled)

**To test:**

Get a file with a spectra and load it into the workbench.
Execute FindPeakAutomatic algorithm and select background type (for gaussian and stretched exponential initial parameter values are necessary most of the times).
Select the other parameters as you want (default values should work reasonably well).
Run the algorithm, a plot of the raw data should appear, with red "x" on all the peaks it found.
A table with the peak parameters should have also been created.

Fixes #26057

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
